package.json

```diff
@@ -11,7 +11,8 @@
   "scripts": {
     "build": "tsc",
     "prepublishOnly": "npm run build",
-    "test": "mocha --parallel && npm run test:tsd",
+    "test-only": "glob -c \"node --loader ts-node/esm --no-warnings --test\" \"test/**/*.spec.ts\"",
```

Comment:
In other project we use tsx (I guess we can replace ts-node)

```ts
glob -c \"tsx --test\" \"./test/**/*.spec.ts\"
```

Reply:
The problem with using tsx to run tests, is that it make them fail because it fail to do some imports. The problem is due to the fact that we use `"type": "module"` inside our `package.json`, removing this fix the tests. The tsx project is aware of the problem, [an issue is open here](https://github.com/privatenumber/tsx/issues/38).

Reply:
I works with tsx.
```bash
npm i -D tsx
```
package.json:
`"test-only": "glob -c \"tsx --test\" \"test/**/*.spec.ts\"",`

src/index.ts
```ts
export { type RC, type Author, JSONSchema, homedir } from "./rc.js";
```
instead of
```ts
export { RC, Author, JSONSchema, homedir } from "./rc.js";
```

src/rc.ts
```ts
import { generateCIConfiguration, type CiConfiguration, type CiWarnings } from "./projects/ci.js";
import { generateReportConfiguration, type ReportConfiguration, type ReportChart } from "./projects/report.js";
import { generateScannerConfiguration, type ScannerConfiguration, type Author } from "./projects/scanner.js";
```
instead of
```ts
import { generateCIConfiguration, CiConfiguration, CiWarnings } from "./projects/ci.js";
import { generateReportConfiguration, ReportConfiguration, ReportChart } from "./projects/report.js";
import { generateScannerConfiguration, ScannerConfiguration, Author } from "./projects/scanner.js";
```

Tested with Node v18, v20, v21

Reply:
Please migrate, we use `tsx` in all our projects